pkg/logqlmodel/metadata/context.go

```diff
@@ -94,3 +119,19 @@ func ExtendHeaders(dst map[string][]string, src []*definitions.PrometheusRespons
 		dst[header.Name] = header.Values
 	}
 }
+
+func AddWarnings(ctx context.Context, warnings ...string) error {
+	context, ok := ctx.Value(metadataKey).(*Context)
```

Comment:
Presumably, most of the time there will be no warnings, and looking up keys in contexts can be surprisingly expensive, not to mention the locking, so a short circuit here is probably a good idea :thinking: 

```suggestion
	if len(warnings) == 0 {
		return nil
	}
	context, ok := ctx.Value(metadataKey).(*Context)
```